contract/contracts/Voting/Types.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+library Types {
+      struct DirectiveVotes {
+        uint256 yes;
+        uint256 no;
+        uint256 abstain;
+    }
+
+    struct Candidate {
+        string name;
+        address candidateAddress;
+        uint256 votes;
+    }
+    struct Member{
+        string name;
+        bool isEligible;
+        bool isVoted;
+        address memberAddress;
+    }   
+    struct Proposal{
+        string title;
+        string description;
+        string draftedBy;
+        bool isElection;
+        bool isActive;
+        DirectiveVotes votes;
+        Candidate[] candidates;
+        Member[] voters;
```

Comment:
i just realized, is it possible that we remove the `DirectiveVotes votes`, so instead we just use `Candidate[] candidates` in candidates we can just create candidate with name "Yes", "No", "Abstain" (dynamic based on the user input) and can be more than 3 choices. and also we can differentiate between election or not based on the value `isElection`

it will looks like this, ignore the naming u can use any naming that you're feel better
```solidity
struct Proposal{
    string title;
    string description;
    string draftedBy;
    bool isElection;
    bool isActive;
    DirectiveVotes votes;
    Choice[] choices;
    mapping(address => bool) voters;
}

struct Choice {
    string name;
    address addressChoice;
    uint256 votes;
}
```

Reply:
As for the Choice struct, good idea but don't you think it is better to separate so that it follows a more generalized approach for the future?
As we add more types of Proposals like Dividend Approvals, ESOP and policy changes, it would be easy to create a structure like struct ESOP{} and then define it inside Proposal as a member.

Reply:
> did u mean like this? https://github.com/farrellh1/cerberus-vault/blob/55e42cbe53f7e03e1291709ff593e255d23df11d/contracts/contracts/CerberusVault.sol#L20

Oh wow. Let me try again then.

Reply:
![image](https://github.com/globe-and-citizen/cnc-portal/assets/31665486/101732ed-52f4-4dfe-827d-f1909cc547f8)

Reply:
> did u mean like this? https://github.com/farrellh1/cerberus-vault/blob/55e42cbe53f7e03e1291709ff593e255d23df11d/contracts/contracts/CerberusVault.sol#L20

![image](https://github.com/globe-and-citizen/cnc-portal/assets/31665486/101732ed-52f4-4dfe-827d-f1909cc547f8)
@farrellh1 great example but look at what happens when I use mapping inside a struct.

Reply:
check dm

Reply:
ohh, dont pass full struct of the proposal, only pass some of important data

Reply:
Also, we cant pass in function parameters as a struct if we have a mapping